unit/java_bytecode/java_virtual_functions/virtual_functions.cpp

```diff
@@ -113,8 +114,63 @@ SCENARIO(
           }
         }
       }
-
       REQUIRE(found_function);
+
+      WHEN("basic blocks are instrumented")
+      {
+        optionst options;
+        options.set_option("cover", "location");
+        THEN("instrumentation assertions are inserted")
+        {
+          REQUIRE(
+            !instrument_cover_goals(
+              options, new_table, new_goto_functions, null_output));
+        }
+        THEN(
+          "instructions are marked as originating from virtual function call "
+          "removal")
+        {
+          auto function = new_goto_functions.function_map.find(function_name);
+          REQUIRE(function != new_goto_functions.function_map.end());
+
+          const goto_programt &goto_program = function->second.body;
+          for(auto it = goto_program.instructions.begin();
```

Comment:
Can this not just be a range based for loop?

Reply:
Or since you are actually just looking for a specific find, perhaps:
```C++
std::find(
  goto_program.instructions.begin(), 
  goto_program.instructions.end(), 
  [](const instructiont &instruction) {
    return instruction.type==goto_program_instruction_typet::GOTO &&
      instruction.guard.id() == ID_equal;
  });
```
This has the advantage you can require that you find the relevant instruction and make it clear you're only looking for the first equal guarded goto.

Reply:
converted to `std::find_if`